server/service/apple_mdm.go

```diff
@@ -3778,8 +3836,37 @@ func preprocessProfileContents(
 						break
 					}
 					hostContents = replaceFleetVariable(fleetVarHostEndUserEmailIDPRegexp, hostContents, emails[0])
+				case FleetVarPKICertPassword:
+					// We just keep it as is for now.
 				default:
-					// This was handled in the above switch statement, so we should never reach this case
+					if strings.HasPrefix(fleetVar, FleetVarPKICertPrefix) {
+						// Get the template name
+						templateName := strings.TrimPrefix(fleetVar, FleetVarPKICertPrefix)
+						if pkiCerts == nil {
+							// This should never happen since we validated the PKI configuration earlier
+							continue
+						}
+						pkiCert, ok := pkiCerts[templateName]
+						if !ok {
+							// This should never happen since we validated the PKI configuration earlier
+							continue
+						}
+						template, ok := pkiTemplates[templateName]
+						if !ok {
+							// This should never happen since we validated the PKI configuration earlier
+							continue
+						}
+						// Insert the new certificate into the profile contents
+						certBase64, err := getNewPKICertificate(ctx, ds, hostUUID, pkiCert, template, appConfig.OrgInfo.OrgName)
+						if err != nil {
+							// This is a server error, so we exit.
+							return ctxerr.Wrap(ctx, err, "getting new PKI certificate")
+						}
+						fleetPKICertRegexp := regexp.MustCompile(fmt.Sprintf(`(\$FLEET_VAR_%s)|(\${FLEET_VAR_%s})`, fleetVar,
+							fleetVar))
+						hostContents = replaceFleetVariable(fleetPKICertRegexp, hostContents, certBase64)
+					}
+					// Invalid variables were handled in the earlier global switch statement, so we should never reach this case
```

Comment:
@dantecatalfamo This is similar code that we would use for $FLEET_SECRET_ variables.